tests/test_input_outputs.py

```diff
@@ -251,3 +254,70 @@ def test_BaseWrapper_postprocess_y():
     np.testing.assert_equal(y_postprocessed, y_array)
     extra_args = BaseWrapper.postprocess_y(y_array)[1]
     assert len(extra_args) == 0
+
+
+@pytest.mark.parametrize(
+    "X_dtype", ["float32", "float64", "int64", "int32", "uint8", "uint16"]
+)
+@pytest.mark.parametrize("y_dtype", ["int64", "int32", "uint8", "uint16"])
```

Comment:
Should we have tests for string/object dtype?

Reply:
Maybe. Right now I'm getting failures when I cast with `y.astype(str)`. It looks like there's some errors in `LabelEncoder` because it assumes the array is sparse.